STAC-23444: add page on ui extension troubleshooting (#109)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/install-stackstate/extension-troubleshooting.adoc

= Extension Troubleshooting
:revdate: 2025-11-03
:page-revdate: {revdate}
:description: SUSE Observability Self-hosted

When you have installed SUSE Observability and the Rancher UI extension, but they do not interoperate as expected, this page may offer some help.

== Collect a HAR (Http ARchive)

> A Http ARchive collects details on requests that are sent by the browser to Rancher and SUSE Observability.  It also logs responses that were received.

Reviewer: ```suggestion
Http ARchive collects details on requests that are sent by the browser to Rancher and {stackstate-product-name}.  It also logs the responses received.
```

This archive is specific to the page/browser tab with the error.  It does not need access to the backing Rancher or SUSE Observability instances (though their logs may be useful in a later stage).

> * Open the "Network" tab in the Developer Tools panel.  This can be accessed by using `F12`.  The network tab shows a log of all request that are made.

Reviewer: ```suggestion
* Open the *Network* tab in the "Developer Tools" panel.  You can access this by using `F12`.  The network tab shows a log of all requests that are made.
```

* Clear the log and do a (hard) refresh of the page that is displaying the problematic behavior.
* There should now be a large number of request to both Rancher and SUSE Observability visible.
* Export/Download/Save the HAR - different browsers use different names here.

== Cluster is unobserved

For the cluster health status to be visible in Rancher, there needs to be alignment between these configurations:

* The (display) name of the cluster in the Rancher UI
* The `stackstate.cluster.name` set on the agent deployed on the cluster
* The cluster name in the configuration of the Kubernetes stackpack in SUSE Observability

A mismatch will tend to show up as "no data received" in the cluster overview in Rancher.

== Connection failed

When the configuration for SUSE Observability does not yield a valid connection in the Rancher UI, this can be due to a number of reasons:

> * The Rancher host is not configured as an allowed origin for SUSE Observability.  This can be remedied by adding it to the `stackstate.allowedOrigins` list in the helm values, or by passing `--set stackstate.allowedOrigins={"https://<rancher-url>"}` to the deployment command.

Reviewer: ```suggestion
* The Rancher host is not configured as an allowed origin for SUSE Observability.  You can fix this by adding it to the `stackstate.allowedOrigins` list in the helm values, or by passing `--set stackstate.allowedOrigins={"https://<rancher-url>"}` to the deployment command.
```
